tc/core/cpu/cpu_tc_executor.cc

```diff
@@ -52,7 +52,8 @@ CpuCompilationResult CpuBackend::compileWithTcMapper(
 
 void CpuTcExecutor::uncheckedRun(
     const std::vector<const void*>& inputs,
-    const std::vector<void*>& outputs) const {
+    const std::vector<void*>& outputs,
+    typename CpuBackend::RuntimeInformation info) const {
```

Comment:
I'm conflicted here:
- by value makes sense because of asynchrony and not wanting to hold reference to local object that may go away. - OTOH the local object is used to launch the kernel synchronously on the CPU so const reference + the above is not a problem for GPUs but may be a problem in the future for other architectures.

So I'm tempted to leave by value, I think this at least deserves some documentation in the .h